### ISSUE TITLE
XLSX Reader: Total cell decreased each row started from line 15

### ISSUE THREAD

**Issue by adakbar** [box/spout#284]
I got inconsistent row iterator with this [file](https://drive.google.com/file/d/0B8IksdS1Vu18eDNQVkJ4cnhKR3M/view?usp=sharing) it should read all 19 cell started from column A, but on line 15, it only get 18 cell, or is there a way that I can set row iterator to read certain column range?

A bit of background, I have a lot Excel file with the same template from client, I tried to count row data that returned by row iterator, interesting part is for every Excel file I received when I read it always decreased to 18 cell that started from line 15.

**Comment by adrilo** [comment]
Hi @adakbar,

I took a look at the file you sent and concluded that the behavior is normal. In the XML file describing the structure of the spreadsheet, here is what I found:

```xml
<row r="1" spans="1:19">
	...
</row>
...
<row r="17" spans="1:18">
	...
</row>
```

Spout parses the `spans` attribute to tell how many cells should be taken into account for the given row. For the first 16th, this number is 19 but it's 18 for the rest. Hence Spout returning 19 values first and only 18 later.

I'm not sure how this file was created but you will either have to re-generate it or add a 19th cell manually in your code.
